From 32594e59991ef4e3f58e250d061c485449b7d0e3 Mon Sep 17 00:00:00 2001
From: Piyush Mehta <piyush.mehta@xilinx.com>
Date: Wed, 15 Jul 2020 12:30:53 +0530
Subject: [PATCH 1445/1851] usb: gadget: udc: Add return value error handling
 of function

commit 54b7373c3b3f83560239a71d39127207151a2e21 from
https://github.com/Xilinx/linux-xlnx.git

This patch handles the function return value.

Reported Coverity warning: CHECKED_RETURN

Signed-off-by: Piyush Mehta <piyush.mehta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/usb/gadget/udc/udc-xilinx.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/udc/udc-xilinx.c b/drivers/usb/gadget/udc/udc-xilinx.c
index 29602c2f6d60..65d47c4979a1 100644
--- a/drivers/usb/gadget/udc/udc-xilinx.c
+++ b/drivers/usb/gadget/udc/udc-xilinx.c
@@ -2206,10 +2206,14 @@ static int xudc_resume(struct device *dev)
 	struct xusb_udc *udc;
 	u32 crtlreg;
 	unsigned long flags;
+	int ret;
 
 	udc = dev_get_drvdata(dev);
 
-	clk_enable(udc->clk);
+	ret = clk_enable(udc->clk);
+	if (ret < 0)
+		return ret;
+
 	spin_lock_irqsave(&udc->lock, flags);
 
 	crtlreg = udc->read_fn(udc->addr + XUSB_CONTROL_OFFSET);
-- 
2.31.1

